### PR TITLE
Fix problem in `xray-lambda` propagator definition

### PR DESCRIPTION
**Pull request by tylerbenson**
the problem comes from this "late" addition:
> To avoid potential issues when extracting with an active span context, the xray-lambda propagator SHOULD check if the provided context already has an active span context. If found, the propagator SHOULD return the provided context unmodified.

This is a problem because `xray-lambda` can't check the original context, only the context that is provided to it, which is modified by prior propagators in the chain.

This means if the original carrier has an x-ray context and an x-ray environment variable set, and if the xray propagator is configured as we expect it to be, then the xray-lambda will be either overwritten (if configured first), or will no-op because it detects an existing span context.

Proposed solution: change `xray-lambda` to replace `xray` in the configured list instead of work together with. This way, it is able to check the context before `xray` modifies it.

## Merge requirement checklist

* [x] [CONTRIBUTING.md](https://github.com/open-telemetry/semantic-conventions/blob/main/CONTRIBUTING.md) guidelines followed.
* [x] Change log entry added, according to the guidelines in [When to add a changelog entry](https://github.com/open-telemetry/semantic-conventions/blob/main/CONTRIBUTING.md#when-to-add-a-changelog-entry).
  * If your PR does not need a change log, start the PR title with `[chore]`
* [x] [schema-next.yaml](https://github.com/open-telemetry/semantic-conventions/blob/main/schema-next.yaml) updated with changes to existing conventions.

### CODE REVIEW

File: docs/faas/aws-lambda.md [hunk 1]
```diff
@@ -70,37 +70,50 @@ Users MUST be able to [configure the propagator](#xray-lambda-propagator-configu
...
-To avoid potential issues when extracting with an active span context, the `xray-lambda` propagator SHOULD check if the provided context already has an active span context. If found, the propagator SHOULD return the provided context unmodified.
-
-Example pseudo implementation:
+This propagator is expected to replace the `xray` propagator in the `OTEL_PROPAGATORS` list. The behavior for this propagator is described in pseudo code below.
 
 ```
 extract(context, carrier) {
+    // To avoid potential issues when extracting with an active span context (such as with a span link),
+    // the `xray-lambda` propagator SHOULD check if the provided context already has an active span context.
+    // If found, the propagator SHOULD just return the extract result of the `xray` propagator.
```
Comment (joaopgrassi) [inline docs/faas/aws-lambda.md:79]:
```suggestion
    // If found, the propagator SHOULD just return the extracted result of the `xray` propagator.
```

File: docs/faas/aws-lambda.md [hunk 2]
```diff
@@ -70,37 +70,50 @@ Users MUST be able to [configure the propagator](#xray-lambda-propagator-configu
...
+- `OTEL_PROPAGATORS=tracecontext,baggage,xray-lambda,xray`
 
-To avoid broken traces, if OpenTelemetry is reporting traces to another system besides AWS X-Ray, users should either omit `xray-lambda` or add it to the beginning:
+**When OpenTelemetry is reporting traces to another system besides AWS X-Ray**, users SHOULD NOT use `xray-lambda` or reported traces will be broken:
 
-`OTEL_PROPAGATORS=xray-lambda,tracecontext,baggage,xray`
+Example valid configuration when OpenTelemetry is reporting traces to another system besides AWS X-Ray:
 
-*Note: The `xray-lambda` propagator can only `extract` context. The `inject` operation MUST be a no-op.*
+- `OTEL_PROPAGATORS=xray-lambda,tracecontext,baggage,xray`
```
Comment (joaopgrassi) [inline docs/faas/aws-lambda.md:116]: Hum above it says:

> **When OpenTelemetry is reporting traces to another system besides AWS X-Ray**, users SHOULD NOT use `xray-lambda` or reported traces will be broken:

Isn't this example here wrong then? 🤔

Comment (tylerbenson) [reply docs/faas/aws-lambda.md:116]: Oops, good catch. Copy paste error.

**Review by jsuereth — APPROVED** [review]

File: docs/faas/aws-lambda.md [hunk 3]
```diff
@@ -70,37 +70,50 @@ Users MUST be able to [configure the propagator](#xray-lambda-propagator-configu
...
+- `OTEL_PROPAGATORS=tracecontext,baggage,xray-lambda`
+
+Example invalid configurations:
 
-`OTEL_PROPAGATORS=tracecontext,baggage,xray,xray-lambda`
+- `OTEL_PROPAGATORS=tracecontext,baggage,xray,xray-lambda`
+- `OTEL_PROPAGATORS=tracecontext,baggage,xray-lambda,xray`
 
-To avoid broken traces, if OpenTelemetry is reporting traces to another system besides AWS X-Ray, users should either omit `xray-lambda` or add it to the beginning:
+**When OpenTelemetry is reporting traces to another system besides AWS X-Ray**, users SHOULD NOT use `xray-lambda` or reported traces will be broken:
```
Comment (trask) [inline docs/faas/aws-lambda.md:112]:
```suggestion
**When OpenTelemetry is reporting traces to another system besides AWS X-Ray**, users SHOULD NOT use `xray-lambda` or reported traces will be broken.
```

**Review by trask — APPROVED** [review]

File: docs/faas/aws-lambda.md [hunk 4]
```diff
@@ -70,37 +70,50 @@ Users MUST be able to [configure the propagator](#xray-lambda-propagator-configu
...
+    context = xrayPropagator.extract(context, carrier)
 
-    traceHeader = getEnvironment("_X_AMZN_TRACE_ID");
+    // If xray-lambda environment variable not set, return.
+    traceHeader = getEnvironment("_X_AMZN_TRACE_ID")
     if (isEmptyOrNull(traceHeader))
       return context
 
+    // Apply the xray propagator using the span context contained in the xray-lambda environment variable.
     return xrayPropagator.extract(context, ["_X_AMZN_TRACE_ID": traceHeader])
```
Comment (rapphil) [inline docs/faas/aws-lambda.md:92]:
```suggestion
    return xrayPropagator.extract(context, ["X-Amzn-Trace-Id": traceHeader])
```

[the remark this answers is not in the document]
Comment (tylerbenson) [reply docs/faas/aws-lambda.md:92]: Good catch.  Thanks.

**Review by lmolkova — APPROVED** [review]

**Review by joaopgrassi — APPROVED** [review]

File: docs/faas/aws-lambda.md [hunk 5]
```diff
@@ -70,37 +70,50 @@ Users MUST be able to [configure the propagator](#xray-lambda-propagator-configu
...
 extract(context, carrier) {
+    // To avoid potential issues when extracting with an active span context (such as with a span link),
+    // the `xray-lambda` propagator SHOULD check if the provided context already has an active span context.
+    // If found, the propagator SHOULD just return the extract result of the `xray` propagator.
     if (Span.fromContext(context).getSpanContext().isValid())
-      return context
+      return xrayPropagator.extract(context, carrier)
+
+    // Update context with the extracted result from xray propagator.
+    context = xrayPropagator.extract(context, carrier)
```
[the remark this answers is not in the document]
Comment (joaopgrassi) [reply docs/faas/aws-lambda.md:84]: Should we address this before merging?

[the remark this answers is not in the document]
Comment (tylerbenson) [reply docs/faas/aws-lambda.md]: I just pushed an update that deduplicates these calls.  LMK if it looks better.

[the remark this answers is not in the document]
Comment (lmolkova) [reply docs/faas/aws-lambda.md]: Thank you!

**Pull request merged**